[🐛 Bug]: Grid endpoint "get downloadable files" can throw IOException

### Description

This problem was introduced in https://github.com/SeleniumHQ/selenium/pull/16589

When using Selenium Grid and calling method `remoteWebDriver.getDownloadableFiles()`, sometimes (very rarely) it may fail. It happens when Chrome downloads its own internal files (e.g. `.com.google.Chrome.GhiE6d`)  - and immediately deletes them.

 

### Reproducible Code

```shell
RemoteWebDriver driver = new RemoteWebDriver();
driver.getDownloadableFiles();
```

### Debugging Logs

```logs
java.io.UncheckedIOException: Failed to get file attributes: /tmp/uuid10303077473612519879bd9e1b7a-d55b-448c-a408-74900b29cc6b/download6602653343356759423/.com.google.Chrome.GhiE6d
	at org.openqa.selenium.grid.node.local.LocalNode.getFileInfo(LocalNode.java:814)
	at org.openqa.selenium.grid.node.local.LocalNode.listDownloadedFiles(LocalNode.java:795)
	at org.openqa.selenium.grid.node.local.LocalNode.downloadFile(LocalNode.java:763)
	at org.openqa.selenium.grid.node.DownloadFile.execute(DownloadFile.java:39)
	at org.openqa.selenium.remote.http.Route.execute(Route.java:69)

Caused by: java.nio.file.NoSuchFileException: /tmp/uuid10303077473612519879bd9e1b7a-d55b-448c-a408-74900b29cc6b/download6602653343356759423/.com.google.Chrome.GhiE6d
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:148)
	at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
	at java.base/java.nio.file.Files.readAttributes(Files.java:1851)
	at org.openqa.selenium.grid.node.local.LocalNode.getFileInfo(LocalNode.java:807)
	... 51 more
```

---
ℹ️ **Last known working version:** `4.38.0`

--- a/java/src/org/openqa/selenium/grid/node/local/LocalNode.java
+++ b/java/src/org/openqa/selenium/grid/node/local/LocalNode.java
@@ -53,6 +53,7 @@ import java.io.Serializable;
 import java.io.UncheckedIOException;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.nio.file.NoSuchFileException;
 import java.nio.file.attribute.BasicFileAttributes;
 import java.nio.file.attribute.FileTime;
 import java.time.Clock;
@@ -792,7 +793,10 @@ public class LocalNode extends Node implements Closeable {
     File[] files = Optional.ofNullable(downloadsDirectory.listFiles()).orElse(new File[] {});
     List<String> fileNames = Arrays.stream(files).map(File::getName).collect(Collectors.toList());
     List<DownloadedFile> fileInfos =
-        Arrays.stream(files).map(this::getFileInfo).collect(Collectors.toList());
+        Arrays.stream(files)
+            .map(this::getFileInfo)
+            .filter(file -> file.getLastModifiedTime() > 0)
+            .collect(Collectors.toList());
 
     Map<String, Object> data =
         Map.of(
@@ -810,6 +814,8 @@ public class LocalNode extends Node implements Closeable {
           attributes.creationTime().toMillis(),
           attributes.lastModifiedTime().toMillis(),
           attributes.size());
+    } catch (NoSuchFileException e) {
+      return new DownloadedFile(file.getName(), -1, -1, -1);
     } catch (IOException e) {
       throw new UncheckedIOException("Failed to get file attributes: " + file.getAbsolutePath(), e);
     }

